ethereum/contracts/bridgehub/bridgehub-deps/BridgehubMailbox.sol

```diff
@@ -0,0 +1,141 @@
+        uint256 _index,
+        L2Message calldata _message,
+        bytes32[] calldata _proof
+    ) external view override returns (bool) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).proveL2MessageInclusion(_batchNumber, _index, _message, _proof);
+    }
+
+    function proveL2LogInclusion(
+        uint256 _chainId,
+        uint256 _batchNumber,
+        uint256 _index,
+        L2Log memory _log,
+        bytes32[] calldata _proof
+    ) external view override returns (bool) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return IStateTransitionChain(stateTransitionChain).proveL2LogInclusion(_batchNumber, _index, _log, _proof);
+    }
+
+    function proveL1ToL2TransactionStatus(
+        uint256 _chainId,
+        bytes32 _l2TxHash,
+        uint256 _l2BlockNumber,
+        uint256 _l2MessageIndex,
+        uint16 _l2TxNumberInBlock,
+        bytes32[] calldata _merkleProof,
+        TxStatus _status
+    ) external view override returns (bool) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).proveL1ToL2TransactionStatus(
+                _l2TxHash,
+                _l2BlockNumber,
+                _l2MessageIndex,
+                _l2TxNumberInBlock,
+                _merkleProof,
+                _status
+            );
+    }
+
+    function l2TransactionBaseCost(
+        uint256 _chainId,
+        uint256 _gasPrice,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit
+    ) external view returns (uint256) {
+        require(address(1) != address(0), "zero addres");
+
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        return
+            IStateTransitionChain(stateTransitionChain).l2TransactionBaseCost(
+                _gasPrice,
+                _l2GasLimit,
+                _l2GasPerPubdataByteLimit
+            );
+    }
+
+    function requestL2Transaction(
+        uint256 _chainId,
+        address _contractL2,
+        uint256 _l2Value,
+        bytes calldata _calldata,
+        uint256 _l2GasLimit,
+        uint256 _l2GasPerPubdataByteLimit,
+        bytes[] calldata _factoryDeps,
+        address _refundRecipient
+    ) public payable override returns (bytes32 canonicalTxHash) {
+        address stateTransitionChain = bridgehubStorage.stateTransitionChain[_chainId];
+        canonicalTxHash = IStateTransitionChain(stateTransitionChain).requestL2TransactionBridgehub(
+            msg.value,
+            msg.sender,
+            _contractL2,
+            _l2Value,
+            _calldata,
+            _l2GasLimit,
+            _l2GasPerPubdataByteLimit,
+            _factoryDeps,
+            _refundRecipient
+        );
+    }
+
+    function finalizeEthWithdrawal(
```

Comment:
We should benchmark how much the ERC20 deposit will cost now. There are quite a bit storage reads/writes/calls with value, etc.